### ISSUE TITLE
Page redirection fully reloads the page

### ISSUE THREAD

**Issue by erickeira** [cloudflare/next-on-pages#70]
Hello, implementing the experimental edge on the CloudFlare page causes the pages to reload, I made texts using route.push and Link, will this be resolved?

**Comment by fineshop** [comment]
I am also facing the same issue.

**Comment by gvatsov** [comment]
Same here! Any comment on this, next-on-pages guys?

**Comment by dario-piotrowicz** [comment]
Hi, I just tested Links and router.push on with the latest next-on-pages version and Next 12.3.1 and it looks like it's working as intended (no full page reload), could you please provide more details about your case or potentially a minimal reproduction? 🙏

**Comment by james-elicx** [comment]
I'm fairly certain this is **not** a next-on-pages issue.

It was one of the bugs in prior versions of Next.js 13 when using the edge runtime. It was fixed last month I believe, although I think the fix might have been unintentional since I never saw a release note mentioning it.

Source: It had been a major blocker for doing anything with the Next.js app directory for me.

**Comment by dario-piotrowicz** [comment]
awesome that's good to hear, thanks 😄:+1:

(I can see you did open an [issue](https://github.com/vercel/next.js/issues/45495) there 🙂)

Let's keep this issue open a bit longer until we test next-on-pages on different Next versions (we're thinking of documenting and testing better what versions of Next we support, so we might just as well close this after double checking then)

**Comment by erickeira** [comment]
I went back to do the tests with next in version 13.2.4 and 12.3.1, locally everything ok, but in the cloudflare page the page always reloads and follows print and my projects that I tested.
https://hotsitenext13.pages.dev/teste3
https://github.com/erickeira/hotsitenext13

Look at the network, it always loads these js files and a css. As no element has changed on the page, there is no css file to load, this shouldn't happen.
![image](https://user-images.githubusercontent.com/11286285/226018775-7c5f3203-8aac-4bce-a6f6-c581300f0120.png)

**Comment by james-elicx** [comment]
Thanks for the reproduction. I've traced the issue back to how `getServerSideProps` appears to work. The page initially requests `/_next/data/.../[page].json`. When this fails, it instead does a hard refresh to the page.

![image](https://user-images.githubusercontent.com/10815538/226029836-892112ef-93a2-495f-adc1-4e1d1dca2f8b.png)

In the `config.json` from the Vercel build output, these locations should be rewritten. next-on-pages doesn't currently support rewrites (#112).

![image](https://user-images.githubusercontent.com/10815538/226030326-f28f5f61-e0b4-41b9-b876-03b1aab64b21.png)

After I tried manually handling the rewrite, there is no hard refresh when clicking the link. I'm afraid this will be a case of having to wait for support for rewrites.
